webapp/package.json

```diff
@@ -23,32 +24,86 @@
     "@babel/preset-react": "7.0.0",
     "@babel/preset-typescript": "7.3.3",
     "@babel/runtime": "7.4.5",
+    "@types/enzyme": "3.10.4",
+    "@types/jest": "^24.0.25",
+    "@types/node": "^12.12.24",
+    "@types/react": "^16.9.17",
+    "@types/react-dom": "^16.9.4",
     "@types/react-intl": "2.3.18",
+    "@types/react-router-dom": "^4.3.4",
+    "@types/react-transition-group": "^4.2.2",
     "@typescript-eslint/eslint-plugin": "1.13.0",
+    "@typescript-eslint/parser": "^1.13.0",
     "babel-eslint": "10.0.2",
+    "babel-jest": "^24.9.0",
     "babel-loader": "8.0.6",
     "babel-plugin-typescript-to-proptypes": "0.17.1",
+    "enzyme": "^3.11.0",
+    "enzyme-adapter-react-16": "^1.15.2",
+    "enzyme-to-json": "^3.4.3",
     "eslint": "6.0.1",
     "eslint-import-resolver-webpack": "0.11.1",
     "eslint-plugin-import": "2.18.0",
     "eslint-plugin-react": "7.14.2",
     "file-loader": "4.0.0",
+    "identity-obj-proxy": "^3.0.0",
     "jest": "^24.8.0",
+    "jest-canvas-mock": "^2.2.0",
+    "jest-junit": "^10.0.0",
     "webpack": "4.35.0",
     "webpack-cli": "3.3.5"
   },
   "dependencies": {
-    "@types/enzyme": "3.10.3",
-    "@types/jest": "24.0.17",
-    "@types/node": "12.7.0",
-    "@types/react": "16.8.24",
-    "@types/react-dom": "16.8.5",
-    "@types/react-router-dom": "4.3.4",
-    "@types/react-transition-group": "4.2.2",
-    "@typescript-eslint/parser": "1.13.0",
+    "core-js": "^3.6.2",
+    "mattermost-webapp": "github:mattermost/mattermost-webapp#23f5f93d9f12a7e2b5623e5cee6814366abd9a0f",
```

Comment:
Do we want all plugins to have a dependency on the webApp? Is this required?

Reply:
It's a devDependency, used only for a one-liner to setup the tests. It's not imported in the plugin's source anywhere.

`import 'mattermost-webapp/tests/setup';`

https://github.com/M-ZubairAhmed/mattermost-plugin-starter-template/blob/4eda4517d417ad9658b7c527109cca698b022cea/webapp/tests/setup.js#L4